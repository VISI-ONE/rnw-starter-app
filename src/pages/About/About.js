import React, { Component } from 'react';
import { StyleSheet, View, Text } from 'react-native';

export class About extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text>About Page</Text>
      </View>
    );
  }
}

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  }
});

export default About;
