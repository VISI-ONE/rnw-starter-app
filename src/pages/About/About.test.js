import React from 'react';
import { View, Text, Platform } from 'react-native';
import { shallow } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import About, { styles } from './About';

describe('About page', () => {
  it('render properly - ' + Platform.OS, () => {
    const tree = renderer.create(<About />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render the proper structure', () => {
    const tree = shallow(<About />);
    expect(
      tree.contains(
        <View style={styles.container}>
          <Text>About Page</Text>
        </View>
      )
    ).toBe(true);
  });
});
