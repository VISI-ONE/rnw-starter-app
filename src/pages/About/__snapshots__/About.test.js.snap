// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page render properly - ios 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    About Page
  </Text>
</View>
`;
